src/cpp/fastdds/subscriber/SubscriberImpl.cpp

```diff
@@ -460,6 +460,50 @@ bool SubscriberImpl::type_in_use(
     return false;
 }
 
+
+void SubscriberImpl::set_qos(
+        SubscriberQos& to,
+        const SubscriberQos& from,
+        bool first_time)
+{
+    if (first_time || !(to.presentation() == from.presentation()))
+    {
+        to.presentation() = from.presentation();
+        to.presentation().hasChanged = true;
+    }
+    if (from.partition().names().size() > 0)
+    {
+        to.partition() = from.partition();
+        to.partition().hasChanged = true;
+    }
+    if (to.group_data().getValue() != from.group_data().getValue() )
+    {
+        to.group_data() = from.group_data();
+        to.group_data().hasChanged = true;
+    }
+    if (to.entity_factory().autoenable_created_entities != from.entity_factory().autoenable_created_entities)
+    {
+        to.entity_factory() = from.entity_factory();
+        to.entity_factory().hasChanged = true;
```

Comment:
```suggestion
```
As this Qos is not going to be sent it is not necessary to set hasChanged to true

Reply:
Then it probably doesn't need to iinherit from QosPolicy either. I'm changing that too.